skills/dff_reasoning_skill/scenario/utils.py

```diff
@@ -72,22 +86,19 @@ def compose_input_for_API(ctx: Context, actor: Actor, *args, **kwargs):
             envvars_to_send,
             **human_uttr_attributes,
         )
-        if len(dialog_context) > 0:
-            try:
-                hypotheses = send_request_to_prompted_generative_service(
-                    dialog_context,
-                    prompt,
-                    GENERATIVE_SERVICE_URL,
-                    GENERATIVE_SERVICE_CONFIG,
-                    GENERATIVE_TIMEOUT,
-                    sending_variables,
-                )
-                api_input = hypotheses[0]
-            except Exception as e:
-                sentry_sdk.capture_exception(e)
-                logger.exception(e)
-                api_input = None
-        else:
+        try:
+            hypotheses = send_request_to_prompted_generative_service(
+                dialog_context,
+                prompt,
+                GENERATIVE_SERVICE_URL,
+                GENERATIVE_SERVICE_CONFIG,
+                GENERATIVE_TIMEOUT,
+                sending_variables,
+            )
+            api_input = hypotheses[0]
+        except Exception as e:
+            sentry_sdk.capture_exception(e)
+            logger.exception(e)
             api_input = None
```

Comment:
please verify all you default values when exception and when non element in shared memory